core/src/avm2/globals/Toplevel.as

```diff
@@ -0,0 +1,9 @@
+package {
+    public namespace AS3 = "http://adobe.com/AS3/2006/builtin";
+
+    public const NaN: Number = 0 / 0;
+
+    public const Infinity: Number = 1 / 0;
+
+    public const undefined = void 0;
```

Comment:
Whoa, this one is new to me! What is the `void 0` syntax? How does this differ from just saying `void`?

Reply:
Ah, I see it is an operator that discards the value here:
https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/operators.html#void
TIL. :)

Reply:
As a value expression, `void` alone is invalid syntax. The [`void` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void), which is supported in JavaScript as well, is pretty useless, apart from providing a way to express `undefined` with less characters (this is a common practice in JS minifiers). Here it is used to allow defining `undefined` in a non-cyclic way, similarly as `avmplus` does: https://github.com/adobe/avmplus/blob/master/core/actionscript.lang.as#L92